src/google/protobuf/stubs/logging.h

```diff
@@ -59,6 +59,19 @@ enum LogLevel {
 #else
   LOGLEVEL_DFATAL = LOGLEVEL_FATAL
 #endif
+
+#ifdef _WIN32
+  // wingdi.h defines ERROR to be 0, so `GOOGLE_LOG(ERROR, ...)` expands
+  // into `GOOGLE_LOG(0, ...)` which then expands into
+  // `someGoogleLogging(LOGLEVEL_0, ...)`. This is not ideal, because the
+  // GOOGLE_LOG macro expects to expand itself into
+  // `someGoogleLogging(LOGLEVEL_ERROR, ...)` instead. The workaround to
+  // get everything building is to simply define LOGLEVEL_0 as
+  // LOGLEVEL_ERROR and also define ERROR the same way that the Windows
+  // SDK does for consistency.
```

Comment:
Would you be able to wrap the wingdi include instead? Something like:
```c++
// safe_wingdi.h
#include <wingdi.h>
#undef ERROR
```
You can then redefine ERROR when all the headers are included in your translation unit.